src/App.tsx

```diff
@@ -7,38 +8,164 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import { User } from './types/User';
+import { getUserPosts, getUsers } from './api/api';
+import { Post } from './types/Post';
 import { Loader } from './components/Loader';
+import { ErrorType } from './types/ErrorType';
+import { ErrorMessage } from './components/ErrorMessage';
+
+class UserInfo implements User {
+  id: number;
+
+  name: string;
+
+  email: string;
+
+  phone: string;
+
+  constructor({
+    id, name, email, phone,
+  }: User) {
+    this.id = id;
+    this.name = name;
+    this.email = email;
+    this.phone = phone;
+  }
+}
+
+class PostInfo implements Post {
+  id: number;
+
+  title: string;
+
+  body: string;
+
+  userId: number;
+
+  constructor({
+    id, title, body, userId,
+  }: Post) {
+    this.id = id;
+    this.title = title;
+    this.body = body;
+    this.userId = userId;
+  }
+}
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [errorType, setErrorType] = useState<ErrorType | null>(null);
+
+  const hasNoPosts = (
+    selectedUser
+    && !userPosts.length
+    && !isLoading
+    && errorType !== ErrorType.POSTS
+  );
+  const hasPosts = (
+    selectedUser
+    && !!userPosts.length
+    && !isLoading
+    && errorType !== ErrorType.POSTS
+  );
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+      const usersInfo = usersFromServer.map((user) => new UserInfo(user));
```

Comment:
why do you create new objects with classes? 

we have ready array with objects from api,  right?